server/lyft/command/feature_runner.go

```diff
@@ -44,7 +46,8 @@ func (r *PlatformModeFeatureRunner) Run(ctx *command.Context, cmd *command.Comme
 		return
 	}
 
-	shouldAllocate, err := r.featureAllocator.ShouldAllocate(feature.PlatformMode, ctx.HeadRepo.FullName)
+	// prCreationTime defaults to time.Unix(0, 0) to ensure features are turned on by default
+	shouldAllocate, err := r.featureAllocator.ShouldAllocate(feature.PlatformMode, ctx.HeadRepo.FullName, time.Unix(0, 0))
```

Comment:
this feels weird that the caller should know this each time. 

Consider a request object and have the featureAllocator itself only care about the default.

Reply:
I added a FeatureContext object which basically has all the metadata needed to determine if a feature is enabled or not. This way, we can also add other attributes in the future to determine the eligibility and not have to worry about passing in the defaults. Lmk what you think!